nosnoc/nosnoc.py

```diff
@@ -945,8 +951,12 @@ def __init__(self, opts: NosnocOpts, model: NosnocModel, ocp: Optional[NosnocOcp
         self.comp_res = Function('comp_res', [self.w, self.p], [J_comp])
         self.g_fun = Function('g_fun', [self.w, self.p], [self.g])
 
+        # copy original w0
+        self.w0_original = copy(self.w0)
```

Comment:
Is this used anywhere?

Reply:
It is rather useful to keep it to reset dual variables to something neutral

Reply:
Indeed the functionality is missing, but I'd like to keep it

Reply:
Got it.